src/memory_region/remote.rs

```diff
@@ -62,20 +63,34 @@ impl RemoteMrWriteAccess for RemoteMr {}
 impl Drop for RemoteMr {
     #[inline]
     fn drop(&mut self) {
-        let agent = Arc::<AgentInner>::clone(&self.agent);
-        let token = self.token;
-        // detach the task
-        let _task = tokio::spawn(async move { AgentInner::release_mr(&agent, token).await });
+        self.agent.as_ref().map_or_else(
+            || debug!("drop a rmr created by user"),
+            |agent| {
+                // this rmr is requested by agent
+                let agent = Arc::<AgentInner>::clone(agent);
+                let token = self.token;
+                // detach the task
+                let _task =
+                    tokio::spawn(async move { AgentInner::release_mr(&agent, token).await });
+            },
+        );
     }
 }
 
 impl RemoteMr {
     /// Create a remote memory region from the `token`
     #[inline]
-    pub(crate) fn new_from_token(token: MrToken, agent: Arc<AgentInner>) -> Self {
+    pub(crate) fn new_from_token(token: MrToken, agent: Option<Arc<AgentInner>>) -> Self {
         Self { token, agent }
     }
 
+    /// Create a remote memory region from the `token` manually
+    #[inline]
+    #[must_use]
+    pub fn new(token: MrToken) -> Self {
```

Comment:
It's from token too, so the name is not proper. I also can't find where the function is called.

Reply:
This is a public API for creating new rmr that without the management of agent. In other words, when the rmr created by this API is going to be dropped, agent willl not send a release msg to the remote end.